### PR TITLE
ovl_Elf_Msg2 decompiled

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 1]
```diff
@@ -31,32 +30,120 @@ const ActorInit Elf_Msg2_InitVars = {
...
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/func_8096EF98.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/func_8096EFD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/ElfMsg2_Update.s")
+void ElfMsg2_SetupAction(ElfMsg2* this, ElfMsg2ActionFunc actionFunc) {
+    (void)"共倒れ";   // Collapse together
+    (void)"共倒れ２"; // Collapse 2
+    (void)"共倒れ";
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:41]: Same comment as in #865

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 2]
```diff
@@ -31,32 +30,120 @@ const ActorInit Elf_Msg2_InitVars = {
...
+        if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
+            Flags_SetSwitch(play, ELFMSG2_GET_8(&this->actor));
+        }
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    if (this->actor.home.rot.y == 0x81) {
+
+        if (Flags_GetClear(play, this->actor.room)) {
+            if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:57]:
```suggestion
        if (Flags_GetClear(play, this->actor.room)) {
            (void)"共倒れ２";
            if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
```
Probably put the 2 here based on OoT

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 3]
```diff
@@ -31,32 +30,120 @@ const ActorInit Elf_Msg2_InitVars = {
...
+    if ((this->actor.home.rot.y > 0) && (this->actor.home.rot.y < 0x81) &&
+        (Flags_GetSwitch(play, this->actor.home.rot.y - 1))) {
+        if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
+            Flags_SetSwitch(play, ELFMSG2_GET_8(&this->actor));
+        }
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    if (this->actor.home.rot.y == 0x81) {
+
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:55]:
```suggestion
    if (this->actor.home.rot.y == 0x81) {
```

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 4]
```diff
@@ -31,32 +30,120 @@ const ActorInit Elf_Msg2_InitVars = {
...
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    return false;
+}
+
+void ElfMsg2_Init(Actor* thisx, PlayState* play) {
+    ElfMsg2* this = THIS;
+
+    if (func_8096EC4C(this, play) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:77]:
```suggestion
    if (!func_8096EC4C(this, play)) {
```

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 5]
```diff
@@ -31,32 +30,120 @@ const ActorInit Elf_Msg2_InitVars = {
...
+        ElfMsg2_SetupAction(this, func_8096EF98);
+        this->actor.flags |= (ACTOR_FLAG_40000 | ACTOR_FLAG_1);
+        this->actor.textId = func_8096EE50(this);
+    }
+}
+
+void ElfMsg2_Update(Actor* thisx, PlayState* play) {
+    ElfMsg2* this = THIS;
+
+    if (func_8096EC4C(this, play) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:146]:
```suggestion
    if (!func_8096EC4C(this, play)) {
```

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c (hunk 1 above)
Comment (sonicdcer) [reply src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:41]: can you give me the exact location of each please?

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 6]
```diff
@@ -31,32 +30,120 @@ const ActorInit Elf_Msg2_InitVars = {
...
+            Flags_SetSwitch(play, ELFMSG2_GET_8(&this->actor));
+        }
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    if (this->actor.home.rot.y == 129) {
+
+        if (Flags_GetClear(play, this->actor.room)) {
+            (void)"共倒れ２"; // "Collapse 2"
+            if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:56]: GET_SWITCH

**Review by hensldm — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**